needy/filesystem.py

```diff
@@ -48,9 +82,30 @@ def lock_file(file_path, timeout=None):
                 fcntl.flock(fd, fcntl.LOCK_EX)
         else:
             fcntl.flock(fd, fcntl.LOCK_EX | fcntl.LOCK_NB)
-        return fd
     except IOError:
-        pass
+        return False
+    return True
+
+
+def lock_fd(fd, timeout=None):
+    '''returns True if the file descriptor is successfully locked'''
+    try:
+        return __win32_lock_fd(fd, timeout)
+    except ImportError:
+        return __fcntl_lock_fd(fd, timeout)
```

Comment:
Are the modules imported here always present on a Windows machine? Will `__fcntl_lock_fd` Raise on Windows?